x-pack/legacy/plugins/siem/public/components/stat_items/index.tsx

```diff
@@ -62,30 +63,43 @@ export interface StatItemsProps extends StatItems {
   barChart?: ChartConfigsData[];
   from: number;
   id: string;
-
   to: number;
+  narrowDateRange: UpdateDateRange;
 }
 
 export const numberFormatter = (value: string | number): string => value.toLocaleString();
-export const areachartConfigs = (from: number, to: number) => ({
+const statItemBarchartRotation: Rotation = 90;
+
+export const areachartConfigs = (config?: {
+  xTickFormatter: (value: number) => string;
+  onBrushEnd?: BrushEndListener;
+}) => ({
   series: {
     xScaleType: ScaleType.Time,
     yScaleType: ScaleType.Linear,
   },
   axis: {
-    xTickFormatter: niceTimeFormatter([from, to]),
+    xTickFormatter: get('xTickFormatter', config),
     yTickFormatter: numberFormatter,
   },
+  settings: {
+    onBrushEnd: getOr(() => {}, 'onBrushEnd', config),
```

Comment:
Reference for `onBrushEnd`:
https://elastic.github.io/elastic-charts/?selectedKind=Interactions&selectedStory=brush%20selection%20tool%20on%20time%20charts&full=0&addons=1&stories=1&panelRight=1&addonPanel=storybook%2Fstories%2Fstories-panel